File: ios/Runner/Native/MessagesChannel.swift
import Flutter
import UIKit

final class MessagesChannel {

    private static let channelName = "pdp.flutter.app/messages"

    private let channel: FlutterBasicMessageChannel

    init(messenger: FlutterBinaryMessenger) {
        channel = FlutterBasicMessageChannel(
            name: Self.channelName,
            binaryMessenger: messenger,
            codec: FlutterStandardMessageCodec.sharedInstance()
        )

        channel.setMessageHandler { message, reply in
            guard let map = message as? [String: Any] else {
                reply([
                    "type": "pong",
                    "id": "unknown",
                    "payload": "Unknown or invalid message",
                ])
                return
            }

            let type = map["type"] as? String
            let id = map["id"] as? String
            let payload = map["payload"] as? String ?? ""

            guard type == "ping", let id, !id.isEmpty else {
                reply([
                    "type": "pong",
                    "id": id ?? "unknown",
                    "payload": "Unknown or invalid message",
                ])
                return
            }

            reply([
                "type": "pong",
                "id": id,
                "payload": "ios:\(payload)",
            ])
        }
    }
}

File: android/app/src/main/kotlin/com/example/pdp_todo_app/native/MessagesChannel.kt
package com.example.pdp_todo_app.native

import io.flutter.plugin.common.BasicMessageChannel
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec

private const val MESSAGES_CHANNEL = "pdp.flutter.app/messages"

class MessagesChannel(
    messenger: BinaryMessenger
) {
    private val channel = BasicMessageChannel(
        messenger,
        MESSAGES_CHANNEL,
        StandardMessageCodec.INSTANCE
    )

    init {
        channel.setMessageHandler { message, reply ->
            val map = message as? Map<*, *>
            val type = map?.get("type") as? String
            val id = map?.get("id") as? String
            val payload = map?.get("payload") as? String ?: ""

            if (type != "ping" || id.isNullOrEmpty()) {
                reply.reply(
                    mapOf(
                        "type" to "pong",
                        "id" to (id ?: "unknown"),
                        "payload" to "Unknown or invalid message"
                    )
                )
                return@setMessageHandler
            }

            reply.reply(
                mapOf(
                    "type" to "pong",
                    "id" to id,
                    "payload" to "android:$payload"
                )
            )
        }
    }
}
